BuildMacOSUniversalBinary.py

```diff
@@ -0,0 +1,278 @@
+
+    # Identity to use for code signing. "-" indicates that the app will not
+    # be cryptographically signed/notarized but will instead just use a
+    # SHA checksum to verify the integrity of the app. This doesn't
+    # protect against malicious actors, but it does protect against
+    # running corrupted binaries and allows for access to the extended
+    # permisions needed for ARM builds
+    "codesign_identity":  '-',
+    # Etitlements file to use for code signing
+    "entitlements": "../Source/Core/DolphinQt/DolphinEmu.entitlements",
+
+    # Minimum macOS version for each architecture slice
+    "arm64_mac_os_deployment_target":  "11.0.0",
+    "x86_64_mac_os_deployment_target": "10.12.0"
+
+}
+# # END CONFIG # #
+
+# Architectures to build for. This is explicity left out of the command line
+# config options for several reasons:
+# 1) Adding new architectures will generally require more code changes
+# 2) Single architecture builds should utilize the normal generated cmake
+#    project files rather than this wrapper script
+
+ARCHITECTURES = ["x86_64", "arm64"]
+
+
+def parse_args(default_conf=DEFAULT_CONFIG):
+    """
+    Parses the command line arguments into a config dictionary.
+    """
+
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+    parser.add_argument(
+        '--target',
+        help='Build target in generated project files',
+        default=default_conf["build_target"],
+        dest="build_target")
+
+    parser.add_argument(
+        '--dst_app',
+        help='Directory where universal binary will be stored',
+        default=default_conf["dst_app"])
+
+    parser.add_argument(
+        '--entitlements',
+        help='Path to .entitlements file for code signing',
+        default=default_conf["entitlements"])
+
+    parser.add_argument(
+        '--codesign',
+        help='Code signing identity to use to sign the applications',
+        default=default_conf["codesign_identity"],
+        dest="codesign_identity")
+
+    for arch in ARCHITECTURES:
+        parser.add_argument(
+             '--{}_pkg_config'.format(arch),
+             help="Folder containing .pc files for {} libraries".format(arch),
+             default=default_conf[arch+"_pkg_config_path"],
+             dest=arch+"_pkg_config_path")
+
+        parser.add_argument(
+             '--{}_qt5_path'.format(arch),
+             help="Install path for {} qt5 libraries".format(arch),
+             default=default_conf[arch+"_qt5_path"])
+
+        parser.add_argument(
+             '--{}_mac_os_deployment_target'.format(arch),
+             help="Deployment architecture for {} slice".format(arch),
+             default=default_conf[arch+"_mac_os_deployment_target"])
+
+    return vars(parser.parse_args())
+
+
+def lipo(path0, path1, dst):
+    if subprocess.call(['lipo', '-create', '-output', dst, path0, path1]) != 0:
+        print("WARNING: {} and {} can not be lipo'd, keeping {}"
```

Comment:
Indices can be used in the format string not to pass `path0` twice.

Reply:
Okay, Ill switch this to use indices to reduce the code size.